llvm/lib/CodeGen/LiveDebugValues/VarLocBasedImpl.cpp

```diff
@@ -2196,24 +2196,28 @@ bool VarLocBasedLDV::ExtendRanges(MachineFunction &MF,
   // Only in the case of entry MBB collect DBG_VALUEs representing
   // function parameters in order to generate debug entry values for them.
   SmallVector<MachineInstr *, 8> AsyncDbgValues;
-  MachineBasicBlock &First_MBB = *(MF.begin());
-  for (auto &MI : First_MBB) {
+  MachineBasicBlock &FirstMBB = *(MF.begin());
+  for (auto &MI : FirstMBB) {
     collectRegDefs(MI, DefinedRegs, TRI);
     if (MI.isDebugValue()) {
       // In Swift async functions entry values are preferred, since they
       // can be evaluated in both live frames and virtual backtraces.
       if (isSwiftAsyncContext(MI)) {
-        MI.getOperand(3).setMetadata(DIExpression::prepend(
-            MI.getDebugExpression(), DIExpression::EntryValue));
+        // If our instruction is not an entry value yet, make it an entry value.
+        if (!MI.getDebugExpression()->isEntryValue()) {
```

Comment:
You'll also need to make the same change in `InstrRefBasedImpl.cpp`.

Reply:
How do I test that?